src/rust/engine/src/fs.rs

```diff
@@ -636,26 +638,42 @@ impl fmt::Debug for Snapshot {
 }
 
 /**
- * A facade for the snapshot directory, which is currently thrown away at the end of each run.
+ * A facade for the snapshot directory, which lives under the pants workdir.
  */
 pub struct Snapshots {
-  temp_dir: TempDir,
-  next_temp_id: atomic::AtomicUsize,
+  snapshots_dir: PathBuf,
+  snapshots_tmpdir: TempDir,
+  next_temp_id: atomic::AtomicUsize
 }
 
 impl Snapshots {
-  pub fn new() -> Result<Snapshots, io::Error> {
+  pub fn new(snapshots_dir: PathBuf) -> Result<Snapshots, String> {
+    let mut snapshots_tmpdir_base = snapshots_dir.clone();
+    snapshots_tmpdir_base.push(".tmp");
```

Comment:
Probably redundant to have `.pants.d/snapshots/.tmp/snapshotsXYZ`... could just do `.pants.d/snapshots/.tmpXYZ`